packages/data-stores/src/site/selectors.ts

```diff
@@ -26,7 +26,13 @@ export const getSite: ( state: State, siteId: number | string ) => SiteDetails |
 		// Try matching numeric site ID
 		state.sites[ siteId ] ||
 		// Then try matching primary domain
-		Object.values( state.sites ).find( ( site ) => site && new URL( site.URL ).host === siteId ) ||
+		Object.values( state.sites ).find( ( site ) => {
+			try {
+				site && new URL( site.URL ).host === siteId;
+			} catch ( error ) {
+				return false;
+			}
+		} ) ||
```

Comment:
The current implementation using try/catch is not the best approach because it hides potential issues. A more efficient way to handle this is by checking if `site.URL` exists before using it, which avoids the error cleanly:

```suggestion
		Object.values( state.sites ).find( ( site ) => site?.URL && new URL( site.URL ).host === siteId ) ||
```

This ensures that we only proceed if site.URL is defined, making the code more readable and performant.